hip-service/Link/Patient/LinkPatient.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection.Metadata.Ecma335;
+using System.Threading.Tasks;
+using hip_service.OTP;
+using HipLibrary.Patient;
+using HipLibrary.Patient.Models;
+using HipLibrary.Patient.Models.Request;
+using HipLibrary.Patient.Models.Response;
+using Optional;
+using CareContext = hip_service.Discovery.Patient.models.CareContext;
+
+namespace hip_service.Link.Patient
+{
+    public class LinkPatient: ILink
+    {
+        private readonly ILinkPatientRepository linkPatientRepository;
+        private readonly PatientRepository patientRepository;
+        private readonly IPatientVerification patientVerification;
+
+        public LinkPatient(ILinkPatientRepository linkPatientRepository, PatientRepository patientRepository,
+            IPatientVerification patientVerification)
+        {
+            this.linkPatientRepository = linkPatientRepository;
+            this.patientRepository = patientRepository;
+            this.patientVerification = patientVerification;
+        }
+
+        public async Task<Tuple<PatientLinkReferenceResponse, ErrorResponse>> LinkPatients(PatientLinkReferenceRequest request)
+        {
+            var patients = patientRepository.GetPatientInfoWithReferenceNumber(request.Patient.ReferenceNumber);
+
+            return await patients.Map(async (patient) =>
+            {
+                var requestPatient = request.Patient;
+
+                var programs = (from careContext in requestPatient.CareContexts
+                    where patientRepository.GetProgramInfo(requestPatient.ReferenceNumber, careContext.ReferenceNumber)
+                        .HasValue
+                    select patientRepository.GetProgramInfo(requestPatient.ReferenceNumber, careContext.ReferenceNumber)
+                    into careContextPatient
+                    select careContextPatient.Map<CareContext>(context =>
+                    {
+                        var newCareContext = new CareContext
+                            {ReferenceNumber = context.ReferenceNumber, Description = context.Description};
+                        return newCareContext;
+                    })).ToList();
+
```

Comment:
nic: Should not mix object convention with sql convention.

Reply:
Fixed